src/callbacks_step/analysis_surface_2d.jl

```diff
@@ -0,0 +1,294 @@
+    @unpack weights = dg.basis
+
+    @unpack indices, variable = surface_variable
+
+    dim = ndims(mesh)
+    n_nodes = nnodes(dg)
+    n_elements = length(indices)
+
+    coords = Matrix{real(dg)}(undef, n_elements * n_nodes, dim) # physical coordinates of indices
+    values = Vector{real(dg)}(undef, n_elements * n_nodes) # variable values at indices
+
+    # Additions for parabolic
+    @unpack viscous_container = cache_parabolic
+    @unpack gradients = viscous_container
+
+    gradients_x, gradients_y = gradients
+
+    surface_integral = zero(eltype(u))
+    index_range = eachnode(dg)
+    global_node_index = 1 # Keeps track of solution point number on the surface
+    for boundary in indices
+        element = boundaries.neighbor_ids[boundary]
+        node_indices = boundaries.node_indices[boundary]
+        direction = indices2direction(node_indices)
+
+        i_node_start, i_node_step = index_to_start_step_2d(node_indices[1], index_range)
+        j_node_start, j_node_step = index_to_start_step_2d(node_indices[2], index_range)
+
+        i_node = i_node_start
+        j_node = j_node_start
+        for node_index in index_range
+            u_node = Trixi.get_node_vars(cache.boundaries.u, equations, dg, node_index,
+                                         boundary)
+
+            x = get_node_coords(node_coordinates, equations, dg, i_node, j_node,
+                                element)
+            # Extract normal direction at nodes which points from the elements outwards,
+            # i.e., *into* the structure.
+            normal_direction = get_normal_direction(direction, contravariant_vectors,
+                                                    i_node, j_node,
+                                                    element)
+
+            gradients_1 = get_node_vars(gradients_x, equations_parabolic, dg, i_node,
+                                        j_node, element)
+            gradients_2 = get_node_vars(gradients_y, equations_parabolic, dg, i_node,
+                                        j_node, element)
+
+            # Integral over whole boundary surface
+            value = variable(u_node, normal_direction, x, t, equations_parabolic,
+                             gradients_1, gradients_2)
+
+            coords[global_node_index, 1] = x[1]
+            coords[global_node_index, 2] = x[2]
+            values[global_node_index] = value
+            i_node += i_node_step
+            j_node += j_node_step
+            global_node_index += 1
+        end
+    end
+    # Sort with ascending x
+    sorting_indices = sortperm(coords[:, 1])
+    coords = coords[sorting_indices, :]
+    values = values[sorting_indices]
+
+    save_pointwise_file(surface_variable.output_directory, varname(variable), coords,
+                        values, t, iter)
+end
+
+varname(::Any) = @assert false "Surface variable name not assigned" # This makes sure default behaviour is not overwriting
```

Comment:
```suggestion
```
Why define a default at all?

Reply:
This is with the intention of giving a helpful error message in case a user forgets to define `varname` for their new surface quantity.

Reply:
I am also not too sure about this one.